polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -1086,32 +1049,51 @@ where
 	let (collator_id, para_id) = peer_data
 		.insert_advertisement(
 			relay_parent,
-			relay_parent_mode,
 			candidate_hash,
 			&state.implicit_view,
 			&state.active_leaves,
+			&per_relay_parent,
 		)
 		.map_err(AdvertisementError::Invalid)?;
 
-	if per_relay_parent.collations.is_seconded_limit_reached(relay_parent_mode) {
+	let claims_for_para = per_relay_parent.collations.claims_for_para(&para_id);
+	let seconded_and_pending_at_ancestors = seconded_and_pending_for_para_in_view(
```

Comment:
We are checking here for our ancestors only. But we could have already seconded collations at the latest leaf.
Suppose A B C relay parents (C being the latest and active)
We second 2 collations on C. Then want to second one on B, we'll only take into account the collations on A (which are none)

Reply:
I built the whole PR under the wrong assumption that relay parents can't 'go backwards'. After talking with you offline and looking through the code again I can see this is not the case.

I'll think about how to fix this.

Reply:
This is now reworked to consider old relay parents and all the paths up to an outer leaf.